torchbenchmark/util/experiment/metrics.py

```diff
@@ -38,19 +40,58 @@ def get_latencies(func, device: str, nwarmup=WARMUP_ROUNDS, num_iter=BENCHMARK_I
         result_summary.append((t1 - t0) / NANOSECONDS_PER_MILLISECONDS)
     return result_summary
 
+def get_peak_memory(func, device: str, num_iter=BENCHMARK_ITERS, export_metrics_file='', metrics_needed=[], metrics_gpu_backend='dcgm') -> Tuple[float, float]:
+    "Run one step of the model, and return the peak memory in MB."
+    from components.model_analyzer.TorchBenchAnalyzer import ModelAnalyzer
+    new_metrics_needed = [_ for _ in metrics_needed if _ in ['cpu_peak_mem', 'gpu_peak_mem']]
+    if not new_metrics_needed:
+        raise ValueError(f"Expected metrics_needed to be non-empty, get: {metrics_needed}")
+    mem_model_analyzer = ModelAnalyzer(export_metrics_file, new_metrics_needed, metrics_gpu_backend)
+    mem_model_analyzer.start_monitor()
+    for _i in range(num_iter):
+        if device == "cuda":
+            torch.cuda.synchronize()
+            func()
+            torch.cuda.synchronize()
+        else:
+            func()
+    mem_model_analyzer.stop_monitor()
+    mem_model_analyzer.aggregate()
+    if 'gpu_peak_mem' in metrics_needed:
+        gpu_peak_mem = mem_model_analyzer.calculate_gpu_peak_mem()
+    else:
+        gpu_peak_mem = None
+    if 'cpu_peak_mem' in metrics_needed:
+        cpu_peak_mem = mem_model_analyzer.calculate_cpu_peak_mem()
+    else:
+        cpu_peak_mem = None
+    if export_metrics_file:
+        mem_model_analyzer.update_export_name("_peak_memory")
```

Comment:
What is this metrics file?

Reply:
`--export-metrics` in `run.py` will export all records to help us debug or find more detailed information.